websites/W/watch.lonelil.com/presence.ts

```diff
@@ -0,0 +1,157 @@
+const presence = new Presence({
+		clientId: "1147910465134002276",
+	}),
+	strings = presence.getStrings({
+		play: "general.playing",
+		pause: "general.paused",
+		viewAMovie: "general.viewAMovie",
+		viewAShow: "general.viewAShow",
+		viewAnime: "general.viewAnime",
+		watchingMovie: "general.watchingMovie",
+		watchingShow: "general.watchingShow",
+		watchingLive: "general.watchingLive",
+		viewingHomePage: "general.viewHome",
+		loading: "kahoot.loadingPage",
+		buttonWatchMovie: "general.buttonWatchMovie",
+		buttonWatchEpisode: "general.buttonViewEpisode",
+		buttonWatchStream: "general.buttonWatchStream",
+		buttonViewMovie: "general.buttonViewMovie",
+		buttonViewShow: "general.buttonViewShow",
+	});
+
+enum logos {
+	logo = "https://i.imgur.com/Niuxx4z.png",
+	home = "https://i.imgur.com/RPr5zwZ.png",
+}
+
+let since = Math.floor(Date.now() / 1000),
+	lastType = "";
+
+presence.on("UpdateData", async () => {
+	if (document.querySelector("#state")) {
+		const state = JSON.parse(
+			document.querySelector("#state").getAttribute("data-state")
+		) as {
+			type: string;
+			since: number;
+			item: {
+				shownType: string;
+				type: "anime" | "movie" | "show" | "tv";
+				id: string;
+				title: string;
+				description: string;
+				poster: string;
+				details?: string;
+			};
+			duration?: number;
+			paused?: boolean;
+			title?: string;
+			shownType?: string;
+		};
+
+		if (state.type !== lastType) {
+			lastType = state.type;
+			since = Date.now();
+		}
+
+		const defaultData: PresenceData = {
+			largeImageKey: logos.logo,
+			smallImageText: "watch.lonelil.com",
+			smallImageKey: logos.logo,
+			startTimestamp: since,
+		};
+		let presenceData: PresenceData = {};
+
+		if (state.type.startsWith("home")) {
+			presenceData = {
+				...defaultData,
+				details: (await strings).viewingHomePage,
```

Comment:
Since strings are used many times throughout the code, it may be cleaner to await the strings once at the start, rather than for every instance.
```ts
// Look at other presences using strings for examples
strings = await strings;
```
```ts
const example = await strings;
example.viewingHomePage;
```